geopyspark/geotrellis/color.py

```diff
@@ -0,0 +1,143 @@
+
+    Returns:
+        [int]
+    """
+
+    if num_colors:
+        return list(geopysc._jvm.geopyspark.geotrellis.ColorRamp.get(ramp_name, num_colors))
+    else:
+        return list(geopysc._jvm.geopyspark.geotrellis.ColorRamp.get(ramp_name))
+
+def get_hex(geopysc, ramp_name, num_colors=None):
+    """Returns a list of the hex values that represent the colors for the given color ramp.
+
+    Note:
+        The returning hex values contain an alpha value.
+
+    Args:
+        ramp_name (str): The name of a color ramp; options are HOT, COOLWARM, MAGMA,
+            INFERNO, PLASMA, VIRIDIS, BLUETOORANGE, LIGHTYELLOWTOORANGE, BLUETORED,
+            GREENTOREDORANGE, LIGHTTODARKSUNSET, LIGHTTODARKGREEN, HEATMAPYELLOWTORED,
+            HEATMAPBLUETOYELLOWTOREDSPECTRUM, HEATMAPDARKREDTOYELLOWWHITE,
+            HEATMAPLIGHTPURPLETODARKPURPLETOWHITE, CLASSIFICATIONBOLDLANDUSE, and
+            CLASSIFICATIONMUTEDTERRAIN
+        num_colors (int, optional): How many colors should be represented in the range. Defaults
+            to ``None``. If not specified, then the full range of values will be returned.
+
+    Returns:
+        [str]
+    """
+
+    if num_colors:
+        return list(geopysc._jvm.geopyspark.geotrellis.ColorRamp.getHex(ramp_name, num_colors))
+    else:
+        return list(geopysc._jvm.geopyspark.geotrellis.ColorRamp.getHex(ramp_name))
+
+class ColorMap(object):
+    def __init__(self, cmap):
+        self.cmap = cmap
+
+    @classmethod
+    def from_break_map(cls, geopysc, break_map, noDataColor=0x00000000, fallback=0x00000000, class_boundary_type=LESSTHANOREQUALTO):
+        if all(isinstance(x, int) for x in break_map.keys()):
+            return cls(geopysc._jvm.geopyspark.geotrellis.ColorMap.fromMap(break_map, noDataColor, fallback, class_boundary_type))
+        elif all(isinstance(x, float) for x in break_map.keys()):
+            return cls(geopysc._jvm.geopyspark.geotrellis.ColorMap.fromMapDouble(break_map, noDataColor, fallback, class_boundary_type))
+        else:
+            raise TypeError("Break map keys must be either int or float.")
+    
+    @classmethod
+    def from_colors(cls, geopysc, breaks, color_list, noDataColor=0x00000000, fallback=0x00000000, class_boundary_type=LESSTHANOREQUALTO):
+        if all(isinstance(x, int) for x in breaks):
+            return cls(geopysc._jvm.geopyspark.geotrellis.ColorMap.fromBreaks(breaks, color_list, noDataColor, fallback, class_boundary_type))
+        else:
+            return cls(geopysc._jvm.geopyspark.geotrellis.ColorMap.fromBreaksDouble([float(br) for br in breaks], color_list, noDataColor, fallback, class_boundary_type))
+
+    @classmethod
+    def from_histogram(cls, geopysc, histogram, color_list, noDataColor=0x00000000, fallback=0x00000000, class_boundary_type=LESSTHANOREQUALTO):
+        pass
+
+    @staticmethod
+    def nlcd_colormap(geopysc):
+        nlcd_color_map =  { 0  : 0x00000000,
```

Comment:
This might be better defined elsewhere so that `nlcd_color_map` isn't being defined each time someone calls `nlcd_colormap`.